File: 3555-final-array-state-after-k-multiplication-operations-i/final-array-state-after-k-multiplication-operations-i.cpp
class Solution {
public:
    vector<int> getFinalState(vector<int>& nums, int k, int multiplier) {
        priority_queue<vector<int>,vector<vector<int>>,greater<vector<int>>> pq;
        
        for(int i = 0;i<nums.size();i++){
            pq.push(vector<int>{nums[i],i});
        }

        for(int i=0;i<k;i++){
            vector<int> res = pq.top();
            pq.pop();
            nums[res[1]] = res[0]*multiplier;
            pq.push(vector<int>{nums[res[1]],res[1] });
        }
        return nums;

    }
};

File: 3555-final-array-state-after-k-multiplication-operations-i/final-array-state-after-k-multiplication-operations-i.java
class Solution {
    public int[] getFinalState(int[] nums, int k, int multiplier) {
        PriorityQueue<int[]> pq = new PriorityQueue<>((a,b)->{
            if(a[0]==b[0]) return a[1]-b[1];
            return a[0]-b[0];
        });
        for(int i = 0;i<nums.length;i++){
            pq.add(new int[]{nums[i],i});
        }

        for(int i = 0;i<k;i++){
            int[] res = pq.poll();
            nums[res[1]] = res[0]*multiplier;
            pq.add(new int[]{nums[res[1]],res[1]});
        }
        return nums;
    }
}